Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -1978,8 +1982,121 @@ - (void)handleCallTap:(TSCall *)call
     [self presentViewController:alertController animated:YES completion:nil];
 }
 
+#pragma mark - MessageActionsDelegate
+
+- (void)messageActionsShowDetailsForItem:(ConversationViewItem *)conversationViewItem
+{
+    [self showDetailViewForViewItem:conversationViewItem];
+}
+
+- (void)messageActionsReplyToItem:(ConversationViewItem *)conversationViewItem
+{
+    [self populateReplyForViewItem:conversationViewItem];
+}
+
+#pragma mark - MenuActionsViewControllerDelegate
+
+- (void)menuActionsDidHide:(MenuActionsViewController *)menuActionsViewController
+{
+    [[OWSWindowManager sharedManager] hideMenuActionsWindow];
+
+    [self updateShouldObserveDBModifications];
```

Comment:
If we're trying to ensure that `updateShouldObserveDBModifications` is called whenever the message actions UI is dismissed, this seems insufficient, since `menuActionsDidHide` is only called when the user initiates the dismissal.  But it should be fine, since the other causes of dismissal (app becoming inactive, incoming call, etc.) should have the side effect of calling  `updateShouldObserveDBModifications`, right?

I just tend to like to ensure invariants in a simpler way - e.g. by having this VC observe a "menu actions was dismissed" event (notification?) that is fired in _all_ cases.

Reply:
e.g. NSNotificationName...isPresentingMenuActionsDidChange

Reply:
Great point. 

To solve, I took a slightly different approach using the delegate, rather than a notification, which I think is a little easier to get right with it's explicit sender/receiver.